### PR TITLE
Enable Lambda Insights for KMS Log Module

### PR DESCRIPTION
**Pull request by henrydrichardson**
### Enable Lambda Insights for KMS Log Module

This PR enables Lambda Insights for KMS Log Module! In turn, this provides [useful performance metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-metrics.html) and memory alarms! Please note, sources for lambda_alerts will be updated to the corresponding  git SHA prior to merging.

_Related PRs:_

- https://github.com/18F/identity-terraform/pull/188
- https://github.com/18F/identity-devops/pull/6572

Partially Addresses #https://github.com/18F/identity-devops/issues/5916

### CODE REVIEW

File: kms_log/main.tf [hunk 1]
```diff
@@ -14,11 +14,17 @@ locals {
 
   lambda_env_variables = merge(local.default_variables, local.alarm_variables)
 
+  lambda_insights = "arn:aws:lambda:${var.region}:580247275435:layer:LambdaInsightsExtension:38"
```
Comment (mww59) [inline kms_log/main.tf:17]: Should this whole arn be a variable? Or maybe add a variable for the version number? It looks like the extension is updated every 2-4 months.

**Review by mww59 — APPROVED** [review]
LGTM

File: kms_log/main.tf (hunk 1 above)
Comment (bleachbyte) [reply kms_log/main.tf:17]: Agreed -- I would go one further and suggest variable-ize-ing the account number, as well, and making a note of what account it refers to (which is unclear from here).

**Review by bleachbyte — CHANGES_REQUESTED** [review]
Check the notes from @mww59 and myself on the `lambda_insights` ARN and give that an update; after that, I'll 👍 this one!

**Comment by henrydrichardson** [comment]
> Agreed -- I would go one further and suggest variable-ize-ing the account number, as well, and making a note of what account it refers to (which is unclear from here).

In this instance the account number is provided by [AWS for all US Regions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-extension-versionsx86-64.html#Lambda-Insights-extension-1.0.143.0). It will remain static irrelevant of target deployment region

File: kms_log/main.tf (hunk 1 above)
Comment (henrydrichardson) [reply kms_log/main.tf]: Requested Changes have been implemented!

**Review by bleachbyte — APPROVED** [review]
Excellent!

**Pull request merged**